lib/base_service.js

```diff
@@ -157,9 +157,17 @@ BaseService.prototype._sanitizeConfig = function(conf, options) {
         // the number of workers has been supplied
         // on the command line, so honour that
         conf.num_workers = options.num_workers;
-    } else if (conf.num_workers === 'ncpu' || typeof conf.num_workers !== 'number') {
-        // use the number of CPUs
-        conf.num_workers = os.cpus().length;
+    }
+    if (typeof conf.num_workers !== 'number') {
+        if (/^ncpu\s*[\*\+\-\/]\s*\d+(?:\.\d+)?$/.test(conf.num_workers)) {
```

Comment:
- Why?
- We already check, that an input string is in a well-defined format with a regex, why checking again?
- You mean `Math.floor(eval(...))`?